FLT/Mathlib/RepresentationTheory/Basic.lean

```diff
@@ -0,0 +1,40 @@
+import Mathlib.LinearAlgebra.Matrix.GeneralLinearGroup.Defs
+import Mathlib.RepresentationTheory.Basic
+
+open LinearMap
+open scoped TensorProduct
+
+namespace Representation
+
+variable {R V G ι: Type*} [CommRing R] [AddCommMonoid V] [Module R V] [Module.Free R V]
+  [Module.Finite R V] [Group G] [DecidableEq ι] [Fintype ι]
+
+variable (ρ : Representation R G V) (𝓑 : Basis ι R V)
+
+omit [Module.Free R V] [Module.Finite R V] in
+@[simp]
+lemma comp_eq_mul (g h : G): ρ g ∘ₗ ρ h = ρ (g * h) := by
+  simp_all only [map_mul]
+  rfl
```

Comment:
The fact that you needed `rfl` after `simp` indicates that there's something wrong here. The goal after `simp_all only [map_mul]` is
```
⊢ ρ g ∘ₗ ρ h = ρ g * ρ h
```
and indeed this is the goal after `simp` (you don't need `simp_all` here, there's nothing in the local context which is helping). The fact that `simp` turns the RHS into `ρ g * ρ h` means that the community has decided that `ρ (g * h)` is not in simp normal form, and that the simp normal form for this expression is `ρ g * ρ h`. So making `comp_eq_mul` into a `simp` lemma is a dubious choice because you are simplifying something into a form which is not the simplest form. Why not
```
@[simp]
lemma comp_def (g h : G): ρ g ∘ₗ ρ h = ρ g * ρ h := rfl
```
? That way you're putting the equation into the form which the community (or at least whoever wrote `map_mul` and tagged it `@[simp]` ) has decided is the simp normal form.